src/FileCacheManager.php

```diff
@@ -86,15 +86,24 @@ public function needFixing($file, $fileContent)
         return false;
     }
 
-    public function setFile($file, $fileContent)
+    public function setFile($file, $fileContent, $isDryRun)
     {
         if (!$this->isCacheAvailable()) {
             return;
         }
 
         $file = $this->getRelativePathname($file);
 
-        $this->newHashes[$file] = $this->calcHash($fileContent);
+        $hash = $this->calcHash($fileContent);
+
+        // in dry run, clear hash for file that needs fixing so it will definitely be inspected on the next run
+        if ($isDryRun && isset($this->oldHashes[$file]) && $this->oldHashes[$file] !== $hash) {
+            unset($this->newHashes[$file]);
```

Comment:
Are we sure the offset `$this->newHashes[$file]` is always set here?
And maybe pass the `dry-run` flag on construction of the `FileCacheManager`?
Anyhow, good idea it will save me lots of processing time as well :+1: